tests/test_object_detection.py

```diff
@@ -0,0 +1,406 @@
+    print(pruned_predictions)
+    for image_pred in pruned_predictions:
+        for class_pred in image_pred:
+            assert class_pred.shape[0] == 0
+
+    pruned_predictions = _prune_by_threshold(predictions, 0.9999)
+
+    num_boxes_not_pruned = 0
+    for image_pred in pruned_predictions:
+        for class_pred in image_pred:
+            if class_pred.shape[0] > 0:
+                num_boxes_not_pruned += 1
+    assert num_boxes_not_pruned == 1
+
+    pruned_predictions = _prune_by_threshold(predictions, 0.0)
+    for im0, im1 in zip(pruned_predictions, predictions):
+        for cl0, cl1 in zip(im0, im1):
+            assert (cl0 == cl1).all()
+
+
+def test_similarity_matrix():
+    ALPHA = 0.99
+    lab_bboxes, lab_labels = _separate_label(labels[0])
+    det_bboxes, det_labels, det_label_prob = _separate_prediction(predictions[0])
+
+    iou_matrix = _get_overlap_matrix(lab_bboxes, det_bboxes)
+    dist_matrix = 1 - _get_dist_matrix(lab_bboxes, det_bboxes)
+
+    similarity_matrix = iou_matrix * ALPHA + (1 - ALPHA) * (1 - dist_matrix)
+    assert (similarity_matrix.flatten() >= 0).all() and (similarity_matrix.flatten() <= 1).all()
+
+
+def test_compute_label_quality_scores():
+    scores = _compute_label_quality_scores(labels, predictions)
+    scores_with_threshold = _compute_label_quality_scores(labels, predictions, threshold=0.9)
+    assert np.sum(scores) != np.sum(scores_with_threshold)
+
+    min_pred_prob = _get_min_pred_prob(predictions)
+    scores_with_min_threshold = _compute_label_quality_scores(
+        labels, predictions, threshold=min_pred_prob
+    )
+    assert (scores == scores_with_min_threshold).all()
+
+
+def test_find_label_issues():
+    alpha = 0.91
+    high_probability_threshold = 0.7
+    low_probability_threshold = 0.1
+
+    thr_overlooked = 0.4
+    thr_badloc = 0.4  # hyperparameter
+    thr_swap = 0.99  # hyperparameter
+
+    auxiliary_inputs = _get_valid_inputs_for_compute_scores(alpha, labels, predictions)
+
+    overlooked_scores_per_box = _compute_overlooked_box_scores(
+        alpha=alpha,
+        high_probability_threshold=high_probability_threshold,
+        auxiliary_inputs=auxiliary_inputs,
+    )
+    overlooked_issues_per_box = _find_label_issues_per_box(
+        overlooked_scores_per_box, thr_overlooked
+    )
+    overlooked_issues_per_image = _pool_box_scores_per_image(overlooked_issues_per_box)
+    overlooked_issues = np.sum(overlooked_issues_per_image)
+    assert overlooked_issues == 3
+
+    badloc_scores_per_box = _compute_badloc_box_scores(
+        alpha=alpha,
+        low_probability_threshold=low_probability_threshold,
+        auxiliary_inputs=auxiliary_inputs,
+    )
+    badloc_issues_per_box = _find_label_issues_per_box(badloc_scores_per_box, thr_badloc)
+    badloc_issues_per_image = _pool_box_scores_per_image(badloc_issues_per_box)
+    badloc_issues = np.sum(badloc_issues_per_image)
+    assert badloc_issues == 1
```

Comment:
assert that the correct ones were detected as well, not just the correct number of them